assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -5542,9 +5550,18 @@ protected void zipGroupSubmissions(String assignmentReference, String assignment
 	    }
 	}
 
-	protected void zipSubmissions(String assignmentReference, String assignmentTitle, String gradeTypeString, int typeOfSubmission, Iterator submissions, OutputStream outputStream, StringBuilder exceptionMessage, boolean withStudentSubmissionText, boolean withStudentSubmissionAttachment, boolean withGradeFile, boolean withFeedbackText, boolean withFeedbackComment, boolean withFeedbackAttachment, boolean withoutFolders,String gradeFileFormat, boolean includeNotSubmitted)
+	protected void zipSubmissions(String assignmentReference, String assignmentTitle, String gradeTypeString, int typeOfSubmission, Iterator submissions, OutputStream outputStream, StringBuilder exceptionMessage, boolean withStudentSubmissionText, boolean withStudentSubmissionAttachment, boolean withGradeFile, boolean withFeedbackText, boolean withFeedbackComment, boolean withFeedbackAttachment, boolean withoutFolders,String gradeFileFormat, boolean includeNotSubmitted, String siteId)
 	{
-	    ZipOutputStream out = null;
+		ZipOutputStream out = null;
+
+		boolean isAdditionalNotesEnabled = false;
+		Site st = null;
+		try{
+			st = SiteService.getSite(siteId);
+			isAdditionalNotesEnabled = candidateDetailProvider.isAdditionalNotesEnabled(st);
+		} catch(Exception e){
```

Comment:
Could we catch the specific `IdUnusedException` here rather than all anything going wrong at all. If it's something more serious we probably want it to blow up and have it logged by something which includes a stack trace.